modules/account-lib/src/coin/dot/proxyBuilder.ts

```diff
@@ -0,0 +1,145 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { decode, methods } from '@substrate/txwrapper-polkadot';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TransactionType } from '../baseCoin';
+import { MethodNames, ProxyArgs, proxyType } from './iface';
+import { ProxyTransactionSchema } from './txnSchema';
+import utils from './utils';
+
+export class ProxyBuilder extends TransactionBuilder {
+  protected _real: string;
+  protected _forceProxyType: proxyType;
+  protected _call: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+  /**
+   *
+   * Dispatch the given call from an account that the sender is authorised for through add_proxy.
+   *
+   * @returns {UnsignedTransaction} an unsigned Dot transaction
+   *
+   * @see https://polkadot.js.org/docs/substrate/extrinsics/#proxy
+   */
+  protected buildDotTxn(): UnsignedTransaction {
+    const baseTxInfo = this.createBaseTxInfo();
+    return methods.proxy.proxy(
+      {
+        real: this._real,
+        forceProxyType: this._forceProxyType,
+        call: this._call,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.Proxy;
+  }
+
+  /**
+   *
+   * The real address of the original tx
+   *
+   * @param {string} real
+   * @returns {ProxyBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#why-use-a-proxy
+   */
+  real(real: string): this {
+    this.validateAddress({ address: real });
+    this._real = real;
+    return this;
+  }
+
+  /**
+   *
+   * The proxy type to execute
+   *
+   * @param {proxyType} forceProxyType
+   * @returns {ProxyBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#proxy-types
+   */
+  forceProxyType(forceProxyType: proxyType): this {
```

Comment:
Why is this needed?

Reply:
Need to specify the exact proxy type of the tx